src/components/token-family/TokenFamilyHeader.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useMemo, useRef } from 'react';
+import { Animated, Easing, StyleSheet, View } from 'react-native';
+import FastImage from 'react-native-fast-image';
+import CaretImageSource from '../../assets/family-dropdown-arrow.png';
+import { useLatestCallback } from '../../hooks';
+import { ButtonPressAnimation } from '../animations';
+import TokenFamilyHeaderIcon from './TokenFamilyHeaderIcon';
+import { Text } from '@rainbow-me/design-system';
+import { ImgixImage } from '@rainbow-me/images';
+import { ThemeContextProps } from '@rainbow-me/theme';
+
+export const TokenFamilyHeaderAnimationDuration = 200;
+export const TokenFamilyHeaderHeight = 50;
+
+const sx = StyleSheet.create({
+  amountContainer: {
+    marginRight: 10,
+  },
+  center: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'center',
+  },
+  chevron: {
+    height: 18,
+    width: 8,
+  },
+  content: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    height: TokenFamilyHeaderHeight,
+    justifyContent: 'space-between',
+    padding: 19,
+    paddingRight: 14,
+    width: '100%',
+  },
+  title: {
+    flex: 1,
+    paddingRight: 9,
+  },
+});
+
+type Props = {
+  childrenAmount?: number;
+  emoji?: string;
+  familyImage?: string;
+  isOpen?: boolean;
+  onPress?: () => void;
+  testID?: string;
+  title: string;
+  theme: ThemeContextProps;
+};
+
+const TokenFamilyHeader = ({
+  childrenAmount,
+  emoji,
+  familyImage,
+  isOpen,
+  onPress,
+  testID,
+  title,
+  theme,
+}: Props) => {
+  const { colors } = theme;
+
+  const toValue = Number(!!isOpen);
+
+  const aRef = useRef<{
+    animation: Animated.Value;
+    toValue: number;
+    isRunning: boolean;
+  } | null>(null);
+
+  // HACK ALERT
+  // we are reusing the same animated value across rows when we recycle them
+  // only happens when previous and next row are both open or closed
+  // but also we don't want the animated value to change
+  // during the animation too
+  if (
+    aRef.current === null ||
+    (aRef.current.toValue !== toValue && !aRef.current.isRunning)
+  ) {
+    aRef.current = {
+      animation: new Animated.Value(toValue),
+      isRunning: false,
+      toValue,
+    };
+  }
+
+  const { animation } = aRef.current!;
+
+  const handlePress = useLatestCallback(() => {
+    Animated.timing(animation, {
+      duration: TokenFamilyHeaderAnimationDuration,
+      easing: Easing.bezier(0.25, 0.1, 0.25, 1),
+      toValue: isOpen ? 0 : 1,
+      useNativeDriver: true,
+    }).start(() => {
+      aRef.current!.isRunning = false;
+    });
+
+    aRef.current!.isRunning = true;
+
+    if (onPress) {
```

Comment:
should we use optional chaining for func too?

Reply:
we could, but personally, i hate these one-liners

```js
onPress?.()
```